src/com/goide/highlighting/GoAnnotator.java

```diff
@@ -80,13 +96,110 @@ else if (element instanceof GoTypeAssertionExpr) {
+      if (args.getType() == null) {
+        if (!args.getExpressionList().isEmpty()) {
+          holder.createErrorAnnotation(call, args.getExpressionList().get(0).getText() + " is not a type");
+        }
+      }
+      else {
+        // We have a type, is it valid?
+        GoType baseType = getBaseType(args.getType());
+        if (!canMakeType(baseType)) {
+          holder.createErrorAnnotation(args.getType(), "Cannot make " + args.getType().getText());
+        }
+        else {
+          // We have a type and we can make the type, are the parameters to make valid?
+          checkMakeArgs(call, args.getType(), baseType, args.getExpressionList(), holder);
+        }
+      }
+    }
+  }
+
+  private static boolean canMakeType(GoType type) {
+    if (type instanceof GoArrayOrSliceType) {
+      // Only slices (no size expression) can be make()'d.
+      return ((GoArrayOrSliceType)type).getExpression() == null;
+    }
+
+    if (type instanceof GoChannelType || type instanceof GoMapType) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private static void checkMakeArgs(GoBuiltinCallExpr call,
+                                    GoType userType,
+                                    GoType baseType,
+                                    List<GoExpression> list,
+                                    AnnotationHolder holder) {
+    if (baseType instanceof GoArrayOrSliceType) {
+      if (list.isEmpty()) {
+        holder.createErrorAnnotation(call, "Missing len argument to make");
+        return;
+      }
+      else if (list.size() > 2) {
+        holder.createErrorAnnotation(call, "Too many arguments to make(" + userType.getText() + ")");
+        return;
+      }
+    }
+
+    if (baseType instanceof GoChannelType || baseType instanceof GoMapType) {
+      if (list.size() > 1) {
+        holder.createErrorAnnotation(call, "Too many arguments to make(" + userType.getText() + ")");
+        return;
+      }
+    }
+
+    for (int i = 0; i < list.size(); i++) {
+      GoExpression expression = list.get(i);
+      GoType expressionType = expression.getGoType(null);
+      if (expressionType != null) {
+        GoType expressionBaseType = getBaseType(expressionType);
+        if (!isIntegerType(expressionBaseType)) {
+          String argName = i == 0 ? "size" : "cap";
+          holder.createErrorAnnotation(expression, "Non-integer " + argName + " argument to make(" + baseType.getText() + ") - " +
+                                                   expressionType.getText());
+        }
+      }
+    }
+  }
+
+  private static GoType getBaseType(GoType type) {
+    if (type.getTypeReferenceExpression() != null) {
+      type = GoPsiImplUtil.findBaseTypeFromRef(type.getTypeReferenceExpression());
+    }
+
+    if (type instanceof GoSpecType) {
+      type = ((GoSpecType)type).getType();
+    }
+
+    return type;
+  }
+
+  private static boolean isIntegerType(GoType type) {
+    return intTypeNames.contains(type.getText());
```

Comment:
Can you give me an example of how I might use it here? For example, in the test I have

```go
type stringType string

make([]stringType, ...)
```

I determine the type of the first argument is a `GoArrayOrSliceType`, calling `getTypeSpecSafe` on that gives me `null`.

I'm unsure how to proceed.